tests/metrics/retrieval/test_map.py

```diff
@@ -0,0 +1,103 @@
+    query_without_relevant_docs_options = ['skip', 'pos', 'neg']
+
+    for batch_size in batch_sizes:
+        for size in sizes:
+            for _ in range(rounds):
+                for behaviour in query_without_relevant_docs_options:
+
+                    metric = torch_class_metric(query_without_relevant_docs=behaviour)
+                    shape = (size,)
+
+                    indexes = []
+                    preds = []
+                    target = []
+
+                    for i in range(batch_size):
+                        indexes.append(
+                            np.ones(shape, dtype=int) * i
+                        )
+                        preds.append(
+                            np.random.randn(*shape)
+                        )
+                        target.append(
+                            np.random.randn(*shape) > 0
+                        )
+
+                    # compute sk metric with multiple iterations using the base `sklearn_metric`
+                    sk_results = []
+                    kwargs = {'device': device, 'dtype': torch.float32}
+                    for b, a in zip(target, preds):
+                        res = sklearn_metric(b, a)
+
+                        if math.isnan(res):
+                            if behaviour == 'skip':
+                                pass
+                            elif behaviour == 'pos':
+                                sk_results.append(
+                                    torch.tensor(1.0, **kwargs)
+                                )
+                            else:
+                                sk_results.append(
+                                    torch.tensor(0.0, **kwargs)
+                                )
+                        else:
+                            sk_results.append(
+                                torch.tensor(res, **kwargs)
+                            )
+                    if len(sk_results) > 0:
+                        sk_results = torch.stack(sk_results).mean()
+                    else:
+                        sk_results = torch.tensor(0.0, **kwargs)
+
+                    indexes = torch.cat([torch.tensor(i) for i in indexes])
+                    preds = torch.cat([torch.tensor(p) for p in preds])
+                    target = torch.cat([torch.tensor(t) for t in target])
+
+                    perm = torch.randperm(indexes.nelement())
+                    indexes = indexes.view(-1)[perm].view(indexes.size())
+                    preds = preds.view(-1)[perm].view(preds.size())
+                    target = target.view(-1)[perm].view(target.size())
+
+                    # shuffle ids to require also sorting of documents ability from the lightning metric
+                    pl_result = metric(indexes, preds, target)
+
+                    assert torch.allclose(sk_results.float(), pl_result.float(), equal_nan=True)
+
+    # check error when `query_without_relevant_docs='error'` is raised correctly
```

Comment:
could you please split these into their own tests as they are checking for valid input

Reply:
@lucadiliello ^^

Reply:
Separated checks on `Exception`s